src/main/java/star/main/Storage.java

```diff
@@ -0,0 +1,77 @@
+package star.main;
+
+import star.exception.StarException;
+import star.task.Deadline;
+import star.task.Event;
+import star.task.Task;
+import star.task.ToDo;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws StarException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new File(filePath);
+            if (!file.exists()) {
+                return tasks;
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String input;
+            while ((input = reader.readLine()) != null) {
+                String[] inputs = input.split(" \\| ");
+
+                Task task;
+                switch (inputs[0]) {
+                case "T":
+                    task = new ToDo(inputs[2]);
+                    break;
+                case "D":
+                    task = new Deadline(inputs[2], inputs[3]);
+                    break;
+                case "E":
+                    task = new Event(inputs[2], inputs[3]);
+                    break;
```

Comment:
Nice use of switch , just thinking that you could add enum for the Task type in the case that you decide to change the icon for each of the task.